### PR TITLE
Added new ISO split layout Layout_60_c_iso, and a keymap example for it for DZ60 board.

### PR DESCRIPTION
**Pull request by Micro-Biology**
## Description

- Added a new iso split layout, very similar to layout_60_b_iso but ?/key amd right shift are swapped.
- Added my new keymap using said layout.

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [x] Keyboard (addition or update)
- [x] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Could you make sure that the info.json contains this layout, as well?  That way, it shows up properly in the Configurator? 
https://docs.qmk.fm/#/reference_configurator_support

If not, that's fine.

**Comment by noroadsleft** [comment]
I looked at this as I was on my way to bed last night, and I just want this documented:

This layout is functionally 100% identical to the `LAYOUT` macro; Assigning `KC_NUHS` to the place where ANSI Backslash goes basically gives you this layout.

I'm also not crazy about the name. `LAYOUT_60_b_ansi` is named such because it's [taken from the DZ60 Plate B](https://github.com/qmk/qmk_firmware/pull/4076), though perhaps I made a mistake in naming it such. Near as I know, there is no Plate C for the DZ60, though perhaps that was an option on the original DZ60 kit.

For the record, this layout is:

- ISO Enter and ISO Left Shift
- Split Backspace
- Split Space
- HHKB-style split Right Shift (1.75u/1u)
- "VIM" bottom right corner, as I call it: 5x 1u keys instead of 4x 1.25u


I'd really like to consolidate the layouts available for the DZ60, TBH. It's already got 13, and this would add another.

\- @noroadsleft
*loves and hates the DZ60*

File: keyboards/dz60/dz60.h [hunk 1]
```diff
@@ -312,7 +312,7 @@
  * │10   │12 │13 │14 │15 │16 │17 │18 │19 │1a │1b │1c │1d │     │
  * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┐2d  │
  * │20    │22 │23 │24 │25 │26 │27 │28 │29 │2a │2b │2c │1e │    │
- * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴┬──┴────┤
+ * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴┬──┴┬───┤
```
Comment (noroadsleft) [inline keyboards/dz60/dz60.h]: Thank you for fixing this. :)

**Comment by Micro-Biology** [comment]
> 
> 
> Could you make sure that the info.json contains this layout, as well? That way, it shows up properly in the Configurator?
> https://docs.qmk.fm/#/reference_configurator_support
> 
> If not, that's fine.

I have updated it on my fork, but I'm new to working on community projects how do I change my pull request to reflect this?



> 
> 
> I looked at this as I was on my way to bed last night, and I just want this documented:
> 
> This layout is functionally 100% identical to the `LAYOUT` macro; Assigning `KC_NUHS` to the place where ANSI Backslash goes basically gives you this layout.
> 
> I'm also not crazy about the name. `LAYOUT_60_b_ansi` is named such because it's [taken from the DZ60 Plate B](https://github.com/qmk/qmk_firmware/pull/4076), though perhaps I made a mistake in naming it such. Near as I know, there is no Plate C for the DZ60, though perhaps that was an option on the original DZ60 kit.
> 
> For the record, this layout is:
> 
>     * ISO Enter and ISO Left Shift
> 
>     * Split Backspace
> 
>     * Split Space
> 
>     * HHKB-style split Right Shift (1.75u/1u)
> 
>     * "VIM" bottom right corner, as I call it: 5x 1u keys instead of 4x 1.25u
> 
> 
> I'd really like to consolidate the layouts available for the DZ60, TBH. It's already got 13, and this would add another.
> 
> - @noroadsleft
> _loves and hates the DZ60_

I totally agree, if I had realised I could have done this I probably would have done so, just to make life easier.

However as it stands I think including this layout is fair as otherwise you could make the argument we don't need any ISO layouts as they could all do this 'hack'. 

A quick question, am I allowed to tidy the DZ60.h file up? And if I did is there a way of giving a layout 2 names, as to not cause incompatibilities with other peoples keymaps?

**Comment by drashna** [comment]
>I have updated the info.json on my fork, but I'm new to working on community projects how do I change my pull request to reflect this?

To clarify, so you understand what/how.... when you open a pull request, it still looks to that branch. Any commit you add and push to github will AUTOMATICALLY update the PR.   In fact, any changes to the branch will update the PR. 
This is one of the reasons we try to dissuade people from using their master branch. (*cough*)

>However as it stands I think including this layout is fair as otherwise you could make the argument we don't need any ISO layouts as they could all do this 'hack'.

Actually, we do need the ISO layout (at least `LAYOUT_60_iso`) for community layout support:

https://docs.qmk.fm/#/feature_layouts?id=supporting-a-layout

Or at least a define to point to it. :)

>A quick question, am I allowed to tidy the DZ60.h file up? And if I did is there a way of giving a layout 2 names, as to not cause incompatibilities with other peoples keymaps?

Nothing to stop you. :)

Though, we generally prefer a couple of PRs for that, than one that tries to do 20 different things.  Though, this .... would be somewhere in between.

**Comment by noroadsleft** [comment]
> I totally agree, if I had realised I could have done this I probably would have done so, just to make life easier.
> 
> However as it stands I think including this layout is fair as otherwise you could make the argument we don't need any ISO layouts as they could all do this 'hack'.

Let me make it plain that this is an issue that's bothered me for a while, and you were just the poor soul who happened to open a DZ60 layout PR.

I did the math some time ago, and there's something like 220 possible layout possibilities for the DZ60. We've had concerns about maintainability, because the "core team" of active contributors is actually pretty small, and it gets to a point where there's just not enough "mental energy" to keep the codebase for a board as popular as the DZ60 well-maintained.

I added the Plate B layouts a while back because they supported some of the "advertised" layouts for the DZ60 on its product page. Even though I'm American, I strongly believe ANSI and ISO should both be supported where possible. I'm just concerned that with a board like the DZ60, every user has their own use case and we can't realistically support everyone's layout.

Most of us who deal with these issues are in agreement about common layouts needing to be supported, like 60% in standard ANSI and ISO, but there still hasn't been a determination of where the line is, past which a layout is too esoteric to need explicit definition.

We recently had a PR where a user was submitting support for a keyboard with 30 possible layouts, but the user went through and painstakingly added every possible layout right off the bat, so there wasn't much of an issue with maintainability. (I did have to PR a slight correction after the fact, but I found the 29 out of 30 success rate impressive. I suspect the user who submitted the keyboard was in direct contact with the PCB's designer.)

> A quick question, am I allowed to tidy the DZ60.h file up? And if I did is there a way of giving a layout 2 names, as to not cause incompatibilities with other peoples keymaps?

Yes, in fact I've done [just that in the past](https://github.com/qmk/qmk_firmware/commits/master/keyboards/dz60?author=noroadsleft).

Obviously you know how the layout macros work, but as it pertains to the Configurator, there's a caveat.

If a user is building from source, you can define a macro and an alias for that macro, and the user can use either name:

```c
#define LAYOUT_ortho_2x3( \
    k00, k01, k02, \
    k10, k11, k12  \
) { \
    { k00, k01, k02 }, \
    { k10, k11, k12 }  \
}
#define LAYOUT LAYOUT_ortho_2x3
```

Here, the user can use `LAYOUT` or `LAYOUT_ortho_2x3` in their keymap, and the result is the same.

Configurator's API though, when you define an alias for a macro, it forgets the macro actually exists, so with the above example, when you define `LAYOUT` at the end, to the Configurator `LAYOUT_ortho_2x3` ceases to exist, and the above to the Configurator looks like:

```c
#define LAYOUT( \
    k00, k01, k02, \
    k10, k11, k12  \
) { \
    { k00, k01, k02 }, \
    { k10, k11, k12 }  \
}
```

... as if `LAYOUT_ortho_2x3` was never defined. Because of that, all `info.json` references to the macro `LAYOUT_ortho_2x3` need to be changed to `LAYOUT`.

For building from source, you can use the alias to modify the arguments:

```c
// Replaces k00 and k10 with a single vertical 2u key on k00
#define LAYOUT_macropad( \
    k00, k01, k02, \
         k11, k12  \
) LAYOUT_ortho_2x3( \
    k00,   k01, k02, \
    KC_NO, k11, k12  \
)
```

... and then as long as the key matrix for `LAYOUT_ortho_2x3` is correct, it'll work.

**Comment by noroadsleft** [comment]
I actually just had an idea:

@drashna, what do you think about merging this basically as is, and then redirecting all the DZ60 layout macros to use the `LAYOUT_all` macro in another PR? That at least helps the maintainability aspect.

Still would need to determine a name for this layout. I still don't like `LAYOUT_60_c_iso` but I'm struggling to come up with something. Actually, @Micro-Biology, why'd you name the macro `LAYOUT_60_c_iso` in the first place? Like, what was your reasoning/thought process?

**Comment by Micro-Biology** [comment]
I just called it that because there was an a and b layout... 

In hindsight, it should be called something else. It is very similar to the ANSI standard layout, but with an iso enter.

Made a new Issue: [Some DZ60 layouts are obsolete. #4668](https://github.com/qmk/qmk_firmware/issues/4668), to discuss consolidating the DZ60 layout marcos.

**Comment by drashna** [comment]
Sorry, I lost track of this issue for a couple of days. 

If you don't mind, I'm going to close this PR, at least until #4668 has a good conclusion.

**Pull request closed without merging**